assets/js/components/Dashboard/PaMessaging/NePaMessagePage/NewPaMessagePage.tsx

```diff
@@ -0,0 +1,145 @@
+import { Button, Container, Row } from "react-bootstrap";
+import WhenCard from "./WhenCard";
+import WhereCard from "./WhereCard";
+import MessageCard from "./MessageCard";
+import fp from "lodash/fp";
+import _ from "lodash";
+import { useNavigate } from "react-router-dom";
+import moment from "moment";
+
+export interface NewPaMessagePageState {
+  selectedStartDate: string;
+  selectedStartTime: string;
+  selectedEndDate: string;
+  selectedEndTime: string;
+  selectedDayLabel: string;
+  selectedDays: number[];
+  selectedPriority: number;
+  selectedInterval: number;
+  visualText: string;
+  phoneticText: string;
+}
+
+export type NewPaMessagePageReducerAction =
+  | {
+      type: "SET_START_DATE";
+      date: string;
+    }
+  | {
+      type: "SET_START_TIME";
+      time: string;
+    }
+  | {
+      type: "SET_END_DATE";
+      date: string;
+    }
+  | {
+      type: "SET_END_TIME";
+      time: string;
+    }
+  | {
+      type: "SET_DAY_LABEL";
+      dayLabel: string;
+    }
+  | {
+      type: "SET_SELECTED_DAYS";
+      days: number[];
+    }
+  | {
+      type: "SET_PRIORITY";
+      priority: number;
+    }
+  | {
+      type: "SET_INTERVAL";
+      interval: number;
+    }
+  | {
+      type: "SET_VISUAL_TEXT";
+      visualText: string;
+    }
+  | {
+      type: "SET_PHONETIC_TEXT";
+      phoneticText: string;
+    };
+
+const reducer = (
+  state: NewPaMessagePageState,
+  action: NewPaMessagePageReducerAction,
+) => {
+  switch (action.type) {
+    case "SET_START_DATE":
+      return fp.set("selectedStartDate", action.date, state);
+    case "SET_START_TIME":
+      return fp.set("selectedStartTime", action.time, state);
+    case "SET_END_DATE":
+      return fp.set("selectedEndDate", action.date, state);
+    case "SET_END_TIME":
+      return fp.set("selectedEndTime", action.time, state);
+    case "SET_DAY_LABEL":
+      return fp.set("selectedDayLabel", action.dayLabel, state);
+    case "SET_SELECTED_DAYS":
+      return fp.set("selectedDays", action.days, state);
+    case "SET_PRIORITY":
+      return fp.set("selectedPriority", action.priority, state);
+    case "SET_INTERVAL":
+      return fp.set("selectedInterval", action.interval, state);
+    case "SET_VISUAL_TEXT":
+      return fp.set("visualText", action.visualText, state);
+    case "SET_PHONETIC_TEXT":
+      return fp.set("phoneticText", action.phoneticText, state);
+  }
+};
+
+const NewPaMessagePage: ComponentType = () => {
+  const now = moment();
+  const initialState: NewPaMessagePageState = {
+    selectedStartDate: now.format("L"),
+    selectedStartTime: now.format("HH:mm"),
+    selectedEndDate: now.format("L"),
+    selectedEndTime: now.add(1, "hour").format("HH:mm"),
+    selectedDayLabel: "All days",
+    selectedDays: [1, 2, 3, 4, 5, 6, 7],
+    selectedPriority: 2,
+    selectedInterval: 4,
+    visualText: "",
+    phoneticText: "",
+  };
+
+  const [state, dispath] = useReducer(reducer, initialState);
```

Comment:
**suggestion:** as there is no logic beyond setting individual keys in the reducer it might actually be simpler to remove the reducer, use a single `useState<NewPaMessagePageState>(initialState)` and update single properties on state in callbacks passed to the children of this component.

Example:

```typescript
onChangeVisualText={(visualText) => 
  setNewPaMessageState(state => ({...state, visualText })}
```

Reply:
Honestly, my reason for going with a reducer is a little silly so I agree that explicit data and callbacks in props is the cleaner way to handle this.